x-pack/plugins/security_solution/cypress/tasks/edit_rule.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CustomRule } from '../objects/rule';
+import {
+  RULE_NAME_HEADER,
+  ABOUT_RULE_DESCRIPTION,
+  ABOUT_STEP,
+  ABOUT_SEVERITY,
+  ABOUT_RISK,
+  RULE_ABOUT_DETAILS_HEADER_TOGGLE,
+  INVESTIGATION_NOTES_TOGGLE,
+  ABOUT_INVESTIGATION_NOTES,
+  INVESTIGATION_NOTES_MARKDOWN,
+  DEFINITION_INDEX_PATTERNS,
+  DEFINITION_STEP,
+  DEFINITION_CUSTOM_QUERY,
+  DEFINITION_TIMELINE,
+  SCHEDULE_STEP,
+  SCHEDULE_RUNS,
+  SCHEDULE_LOOPBACK,
+} from '../screens/rule_details';
+import { EDIT_SUBMIT_BUTTON } from '../screens/edit_rule';
+
+export const expectRuleDetails = (rule: CustomRule) => {
+  const expectedTags = rule.tags.join('');
+  const expectedIndexPatterns =
+    rule.index && rule.index.length
+      ? rule.index
+      : [
+          'apm-*-transaction*',
+          'auditbeat-*',
+          'endgame-*',
+          'filebeat-*',
+          'logs-*',
+          'packetbeat-*',
+          'winlogbeat-*',
+        ];
+
+  cy.get(RULE_NAME_HEADER).invoke('text').should('eql', `${rule.name} Beta`);
+
+  cy.get(ABOUT_RULE_DESCRIPTION).invoke('text').should('eql', rule.description);
+  cy.get(ABOUT_STEP).eq(ABOUT_SEVERITY).invoke('text').should('eql', rule.severity);
+  cy.get(ABOUT_STEP).eq(ABOUT_RISK).invoke('text').should('eql', rule.riskScore);
+  cy.get(ABOUT_STEP).eq(2).invoke('text').should('eql', expectedTags);
```

Comment:
@MadameSheema due to the dynamic nature of our Rule Details views, I found that these hardcoded indexes (`ABOUT_SEVERITY`, `RULE_TYPE`, etc.) would only work for rules with specific data (i.e. rules created manually). For the rules in the `custom_rules` archive, many of these fields don't exist and so these indexes don't work (e.g. `ABOUT_TAGS = 5` but there are only three fields for a custom rule (severity, risk_score, and tags).

While I can modify the `custom_rules` archive to include the data necessary to make this work, I was wondering if you had any other thoughts here: setting `data-test-subj` on these fields and querying them directly, or adding some kind of logic to dynamically determine the indexes both seem like less brittle approaches.

Would you recommend

Reply:
If is possible the best solution is to add `data-test-subj` to all the elements 😊 

Can you please check if it can be done? If not we can think in a different solution.

Reply:
@MadameSheema adding `data-test-subj` didn't seem feasible as `EuiDescriptionList` doesn't currently support its children having such a prop, so I went the "dynamic indexes" route, which I think turned out well: https://github.com/elastic/kibana/pull/77090/commits/569980b2928e10c67bcdff76761e41a1cbc1b67b